#167323728 user should be able to receive email notifications

timi-codes opened the pull request:
#### Description
 - Users  that opted-in for email notification can get email notifications on being followed 
 - An Author that opted-in for email notification can get email notifications when there is a new comment on their article.
 - All followers that opted-in for email notification get an email notification when they publish a new article
#### Type of Change

- [x] New feature (Non-breaking change which adds functionality)

- [x]  Bug fix (Non-breaking change which adds functionality)

#### How Has This Been Tested?
  Integration Testing
#### Pivotal Stories
[#167323728](https://www.pivotaltracker.com/story/show/167323728)

Onyimatics reviewed (approved):
Good work @timi-codes and @codexempire. I like the way you named your variables. Looks good to me.

tohbay reviewed (changes requested):
Looks good to me 👍 

Resolve the merge conflicts

timi-codes committed:
ft(notification): user can opt in and get email notification
- when they they are followed
- when there is a new comment on their article
- when a followed author published an article
[Delivers #167323728]

kechyy reviewed (approved):
Good work @timi-codes . This looks good to me.
